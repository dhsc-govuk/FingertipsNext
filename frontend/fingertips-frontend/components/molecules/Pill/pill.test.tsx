import { render, screen } from '@testing-library/react';
import { Pill } from '.';
import { userEvent } from '@testing-library/user-event';

const mockFilterRemoveFunction = jest.fn();
const selectedFilterName = 'Dementia';
const selectedFilterId = '001';
const handleSelectedFilterRemove = mockFilterRemoveFunction;

describe('Pill Suite', () => {
  it('should render expected elements', () => {
    render(
      <Pill
        selectedFilterName={selectedFilterName}
        selectedFilterId={selectedFilterId}
        handleSelectedFilterRemoval={handleSelectedFilterRemove}
      />
    );

    expect(screen.getByTestId('pill-container')).toBeInTheDocument();
    expect(screen.getByTestId('filter-name')).toBeInTheDocument();
    expect(screen.getByTestId('remove-icon-div')).toBeInTheDocument();
    expect(screen.getByRole('paragraph')).toBeInTheDocument();
    expect(screen.getByTestId('x-icon')).toBeInTheDocument();
  });

  it('should render text passed in as prop', () => {
    render(
      <Pill
        selectedFilterName={selectedFilterName}
        selectedFilterId={selectedFilterId}
        handleSelectedFilterRemoval={handleSelectedFilterRemove}
      />
    );

    expect(screen.getByRole('paragraph')).toHaveTextContent('Dementia');
  });

  it('should call the passed in handle function when clicking the remove icon', async () => {
    const user = userEvent.setup();
    mockFilterRemoveFunction.mockClear();

    render(
      <Pill
        selectedFilterName={selectedFilterName}
        selectedFilterId={selectedFilterId}
        handleSelectedFilterRemoval={handleSelectedFilterRemove}
      />
    );

    await user.click(screen.getByTestId('remove-icon-div'));

    expect(mockFilterRemoveFunction).toHaveBeenCalledWith(selectedFilterId);
  });

  it('snapshot test', () => {
    const container = render(
      <Pill
        selectedFilterName={selectedFilterName}
        selectedFilterId={selectedFilterId}
        handleSelectedFilterRemoval={handleSelectedFilterRemove}
      />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
