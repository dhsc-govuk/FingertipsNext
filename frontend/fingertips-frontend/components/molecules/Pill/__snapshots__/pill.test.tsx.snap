// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill Suite snapshot test 1`] = `
<DocumentFragment>
  .c2 {
  cursor: pointer;
}

.c0 {
  background-color: white;
  border: 1px #D1D2D3 solid;
  border-radius: 5px;
  padding: 0.3125em 0.3125em;
  max-width: 100%;
  margin: 0.3125em 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  word-wrap: break-word;
  padding-left: 0.5em;
  font-size: 16px;
  margin-bottom: 0;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    class="c0"
    data-testid="pill-container"
  >
    <div
      class="c1"
      data-testid="remove-icon-div"
    >
      <svg
        class="c2"
        data-testid="x-icon"
        fill="none"
        height="12"
        stroke="#000000"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="3"
        viewBox="0 0 25 25"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="c3"
      data-testid="filter-name"
    >
      Dementia
    </div>
  </div>
</DocumentFragment>
`;
